app/stac_api/models.py

```diff
@@ -543,7 +546,8 @@ def delete(self, *args, **kwargs):  # pylint: disable=signature-differs
             raise ValidationError(error.args[0]) from None
 
     def clean(self):
-        validate_asset_name_with_media_type(self.name, self.media_type)
+        media_type = validate_and_get_media_type(self.name, self.media_type)
```

Comment:
I'm not sure anymore, but I think this method `validate_and_get_media_type()` could be simplified and should not be needed here.
The `clean()` method is used for validation (also clean is a bad word for this but comes from django) of fields dependencies. I think we should separate the media type validation from the the media type and asset name dependency validation. I mean leaving here the validation of the asset name with the media type and add a media type validator in the field for checking the media type validity.
This way you would ensure that the media type in DB is always a valid normalized string.